src/main/scala/coreplex/CrossingWrapper.scala

```diff
@@ -0,0 +1,96 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.coreplex
+
+import Chisel._
+import freechips.rocketchip.config._
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.util._
+
+/** Enumerates the three types of clock crossing between tiles and system bus */
+sealed trait CoreplexClockCrossing
+case class SynchronousCrossing(params: BufferParams = BufferParams.default) extends CoreplexClockCrossing
+case class RationalCrossing(direction: RationalDirection = FastToSlow) extends CoreplexClockCrossing
+case class AsynchronousCrossing(depth: Int, sync: Int = 3) extends CoreplexClockCrossing
+
+trait HasCrossingHelper extends LazyScope
+{
+  this: LazyModule =>
+  val crossing: CoreplexClockCrossing
+
+  def cross(x: TLCrossableNode): TLOutwardNode = {
+    val out = x.node.parentsOut.exists(_ eq this) // is the crossing exiting the wrapper?
+    crossing match {
+      case SynchronousCrossing(params) => {
+        // !!! Why does star resolution fail for tile with no slave devices?
+        // this { TLBuffer(params)(x.node) }
```

Comment:
When a tile has 0 slave devices (e.g. `DefaultConfig`), `resolveStar` fails here and I cannot figure out why. I though I was missing a `:*=` but I am pretty confident `SinkCardinality` is applied by `pbus.toTile` to all these adapter/crossing functions. The below line works; it's only when a buffer with :*= on both sides of it (I think) is inserted...

Any chance you could poke at it a little? It's driving me crazy.